bftengine/src/bcstatetransfer/BCStateTran.cpp

```diff
@@ -1438,14 +1439,38 @@ uint16_t BCStateTran::getBlocksConcurrentAsync(uint64_t nextBlockId, uint64_t fi
   return j;
 }
 
-void BCStateTran::srcInitialize() {
+void BCStateTran::initializeSource(uint16_t replicaId) {
   // a new source - reset histograms and snapshot counter
-  sourceFlag_ = true;
+  sourceSession_.setNew(replicaId);
   auto &registrar = concord::diagnostics::RegistrarSingleton::getInstance();
   registrar.perf.snapshot("state_transfer");
   registrar.perf.snapshot("state_transfer_src");
 }
 
+bool BCStateTran::shouldContinue(uint16_t replicaId) {
```

Comment:
I think this function should not be one of `BCStateTran`'s member functions. I suggest moving it to the `SourceSession` class. The source can be refactored around this change.
This is just a suggestion, however if it won't be applied I think a better and more descriptive name should be given to this function.